test/e2e/test-basic.ts

```diff
@@ -2,16 +2,122 @@ import { assert } from 'chai'
+    it('should change the size of the filter array when removing a filter', function() {
+      let ctrlZ = httpClient.addFilter(aFilter, "the first filter")
+      const nbOfFiltersBeforeDelete = (httpClient as any)._filters.length
+      ctrlZ.remove()
+      const nbOfFiltersAfterDelete = (httpClient as any)._filters.length
+      assert.equal(nbOfFiltersAfterDelete, nbOfFiltersBeforeDelete - 1)
+    })
+    it('should not remove a filter when adding one', function() {
+      let ctrlZ = httpClient.addFilter(aFilter, 'the first filter')
+      let ctrlZZ = httpClient.addFilter(anotherFilter, 'the second filter')
+      function firstFilterStillThere(element: httpclient.InstalledFilter) {
+        return element.name === 'the first filter';
+      }
+      assert.notEqual((httpClient as any)._filters.findIndex(firstFilterStillThere), -1);
+    })
+    it('should not remove the first filter when removing the second filter ', function() {
+      httpClient.addFilter(aFilter, 'the first filter')
+      let ctrlZZ = httpClient.addFilter(anotherFilter, 'the second filter')
+      ctrlZZ.remove()
+      assert.equal((httpClient as any)._filters[0].name, "the first filter")
+    })
+  })
+  describe('call', function() {
+    it('should return the body of the called request', async function() {
+      const body = await httpClient.call<Object>(aRequest)
+      assert.deepEqual(body, aResponse.body)
+    })
+  })
+  describe('callForResponse' , function() {
+    it('should return a Response object', async function() {
+      const theResponse: httpclient.Response<Object> = await httpClient.callForResponse<Object>(aRequest)
+      assert.instanceOf(theResponse, httpclient.Response)
+    })
+  })
+  describe('set', function() {
+    it('should change all the proprities of a Request', function() {
+      let partialRequest: {
+        contentType?: string, method?: 'GET' | 'POST' | 'PUT' | 'DELETE' | string,
+        responseType?: XMLHttpRequestResponseType,
+        withCredentials?: boolean, 
+        body?: object | Document | BodyInit | null, 
+        headers?: httpclient.Headers, 
+        timeout?: number
+      } = {}
+      partialRequest.contentType = 'application/json;'
+      partialRequest.method = 'DELETE'
+      partialRequest.responseType = 'text'
+      partialRequest.withCredentials = true
+      partialRequest.body = {}
+      partialRequest.headers = {'aKey': 'aValue'}
+      partialRequest.timeout = 20000 
+      aRequest.set(partialRequest)
+      // Test successful only if each proprety has changed
+      assert.notEqual(aRequest.contentType, 'application/json; charset=UTF-8')
+      assert.notEqual(aRequest.method, 'GET')
+      assert.notEqual(aRequest.responseType, 'json')
+      assert.notEqual(aRequest.withCredentials, false)
+      assert.notEqual(aRequest.body, null)
+      assert.notEqual(aRequest.headers, {})
+      assert.notEqual(aRequest.timeout, 30000)
+    })
+  })
+  // describe('what to describe', function() {
```

Comment:
cloud you add a test to verify that filters are actually called on the response?